R/sim_gs_n.R

```diff
@@ -241,6 +250,21 @@
 #'   weight = fh(rho = 0, gamma = 0)
 #' )
 #' plan("sequential")
+#'
+#' # Example 10: group sequential design with updated bounds
+#' x <- gs_design_ahr(analysis_time = 1:3*12) |> to_integer()
+#' sim_gs_n(
+#'   n_sim = 3,
+#'   sample_size = max(x$analysis$n),
+#'   enroll_rate = x$enroll_rate,
+#'   fail_rate = x$fail_rate,
+#'   test = wlr,
+#'   cut = list(ia1 = create_cut(planned_calendar_time = x$analysis$time[1]),
+#'              ia2 = create_cut(planned_calendar_time = x$analysis$time[2]),
+#'              fa = create_cut(planned_calendar_time = x$analysis$time[3])),
+#'   weight = fh(rho = 0, gamma = 0),
+#'   original_design = x
+#' )
```

Comment:
We may need to start using `\donttest{}` for some of these examples. The `R CMD check` results identify the `sim_gs_n()`
examples as taking more than 5 seconds:

```R
* checking examples ... [18s/19s] OK
Examples with CPU (user + system) or elapsed time > 5s
          user system elapsed
sim_gs_n 5.016  0.365   6.627
* checking examples with --run-donttest ... [18s/21s] OK
```

This note didn't appear in the [CI runs](https://github.com/Merck/simtrial/actions/runs/14885038375) for the most recent commit to main.

Surprisingly though, when I run this new example on my local machine, it is very fast.

We can wait to address this until immediately prior to the CRAN submission if you like. Especially since we are optimizing the {gsDesign2} code to be faster.

Reply:
Thank you, John! I currently reduce the number of simulations `n_sim` to 1. Since the example is just for demo purpose, I guess 1 simulation should be good. If the CRAN asks, we can `dontest{}` this example when do a CRAN release!